internal/action/config.go

```diff
@@ -89,10 +89,29 @@ func (s *Action) setConfigValue(ctx context.Context, store, key, value string) e
 		return fmt.Errorf("storage not available")
 	}
 
-	if err := st.Add(ctx, "config"); err != nil && !errors.Is(err, istore.ErrGitNotInit) {
+	// in case of a local config change we want to track changes
+	if !st.Exists(ctx, "config") {
```

Comment:
This seems a little bit hacky. Are you doing this only because s.cfg.Set doesn't tell you if it wrote to a global or local config? In that case we could (should) add an additional return value to config.Set to make this easier.

Reply:
Yes, I am. I'm happy adding a "iota enum" going in maybe reverse order from 0 to x in order of precedence of the configs?
Lower numbers having less precedence than higher ones. (e.g. Env would be the max value of the top of my head)

Reply:
But this specific line is to avoid attempting to interact with a non existent local config file since that causes git to error out.

Reply:
The `Set` method is part of the internal package `internal/config`. Feel free to add a second parameter if you need it.

But could you remind me how/where you are planning to use that? Right now I fail to see how that would help this PR.